lms/djangoapps/teams/tests/test_models.py

```diff
@@ -73,3 +85,50 @@ def test_user_in_team_for_course(self, username, course_id, expected_value):
             CourseTeamMembership.user_in_team_for_course(user, course_id),
             expected_value
         )
+
+
+@ddt.ddt
+class TeamSignalsTest(SharedModuleStoreTestCase):
+    """Tests for handling of team-related signals."""
+
+    def setUp(self):
+        """Create a user with a team to test signals."""
+        super(TeamSignalsTest, self).setUp()
+        self.user = UserFactory.create(username="user")
+        self.moderator = UserFactory.create(username="moderator")
+        self.team = CourseTeamFactory(course_id=COURSE_KEY1)
+        self.team_membership = CourseTeamMembershipFactory(user=self.user, team=self.team)
+
+    @ddt.data(
+        *itertools.product(
+            (
+                thread_created,
+                thread_edited,
+                thread_voted,
+                comment_created,
+                comment_edited,
+                comment_voted,
+                comment_endorsed
+            ),
+            (('user', True), ('moderator', False))
+        )
+    )
+    @ddt.unpack
+    def test_signals(self, signal, (user, should_update)):
```

Comment:
Note: there's probably some way to make this DDT data look nicer without needing to use tuple unpacking in the argument list, but I'm not sure exactly how.